projects/odf_transform/odf_transform/odf_parser/odf_parser.py

```diff
@@ -165,68 +171,115 @@ def _convert_to_number(value):
     return metadata, data_raw
 
 
-def define_odf_variable_attributes(metadata,
-                                   organizations=None,
-                                   vocabulary=None,
-                                   vocabulary_attribute_list=None,
-                                   odf_var_header_prefix='original_',
-                                   odf_variable_name="CODE",
-                                   flag_prefix='QualityFlag:'):
+def odf_flag_variables(metadata, flag_convention=None):
+    """
+    odf_flag_variables handle the different conventions used within the ODF files over the years and map them
+     to the CF standards.
+    """
+
+    # Loop through each variables and detect flag variables
+    previous_key = None
+    for var, att in metadata.items():
+        # Retrieve information from variable name
+        odf_var_name = parse_odf_code_variable(var)
+        related_variable = None
+
+        # FLAG VARIABLES Detect if it is a flag column
+        is_q_flag = var.startswith("Q") and var[:1] in metadata.keys()
+        is_qqqq_flag = odf_var_name['name'] == 'QQQQ'
+        is_general_flag = odf_var_name['name'] in ['QCFF', 'FFFF']
+        is_flag_column = is_q_flag or is_qqqq_flag or is_general_flag
+
+        # Find related variable
+        if is_qqqq_flag:
+            # MLI FLAG QQQQ should apply to previous variable
+            related_variable = previous_key
+        elif is_q_flag:
+            # Q  Format is usually Q+[PCODE] of the associated variable
+            related_variable = var[1:]
+            # Make sure that related_variable do exist!
+            if related_variable not in metadata:
```

Comment:
looks like indent might be off here, not sure

Reply:
this must be outdated with all the recent changes